beacon/validator/block_builder.go

```diff
@@ -312,45 +311,23 @@ func (s *Service[
 		return ErrNilDepositIndexStart
 	}
 
-	// Bartio and Boonet pre Fork2 have deposit broken and undervalidated
-	// Any other network should build deposits the right way
-	if !(s.chainSpec.DepositEth1ChainID() == spec.BartioChainID ||
-		(s.chainSpec.DepositEth1ChainID() == spec.BoonetEth1ChainID &&
-			blk.GetSlot() < math.U64(spec.BoonetFork2Height))) {
-		depositIndex++
-	}
-	blkDeposits, err := s.sb.DepositStore().GetDepositsByIndex(
-		depositIndex,
-		s.chainSpec.MaxDepositsPerBlock(),
+	// Grab all previous deposits from genesis up to the current index + max deposits per block.
+	deposits, err := s.sb.DepositStore().GetDepositsByIndex(
+		0, depositIndex+s.chainSpec.MaxDepositsPerBlock(),
 	)
 	if err != nil {
 		return err
 	}
 
-	// Set the deposits on the block body.
+	var eth1Data *ctypes.Eth1Data
+	body.SetEth1Data(eth1Data.New(deposits.HashTreeRoot(), 0, common.ExecutionHash{}))
+
+	// Set the just the block deposits (after current index) on the block body.
 	s.logger.Info(
 		"Building block body with local deposits",
-		"start_index", depositIndex, "num_deposits", len(blkDeposits),
-	)
-	body.SetDeposits(blkDeposits)
-
-	// Grab all previous deposits from genesis up to the current index
-	// to calculate deposit root.
-	deposits, err := s.sb.DepositStore().GetDepositsByIndex(
-		0,
-		depositIndex,
+		"start_index", depositIndex, "num_deposits", uint64(len(deposits))-depositIndex,
 	)
-	if err != nil {
-		return err
-	}
-	deposits = append(deposits, blkDeposits...)
-
-	var eth1Data *ctypes.Eth1Data
-	body.SetEth1Data(eth1Data.New(
-		ctypes.Deposits(deposits).HashTreeRoot(),
-		0,
-		common.ExecutionHash{},
-	))
+	body.SetDeposits(deposits[depositIndex:])
```

Comment:
`GetDepositsByIndex` could theoretically return a slice of length less than `depositIndex`, for example if there is a missing index anywhere in the depositStore (which probably shouldn't happen, but it *possibly could*). There should be an index check before indexing into this to prevent panic in this special case.

Reply:
done